src/containers/MyNdla/components/SettingsMenu.tsx

```diff
@@ -193,9 +193,12 @@ const SettingsMenu = ({ menuItems, modalHeader }: Props) => {
                         colorTheme={item.type}
                         ref={item.ref}
                         onClick={(e) => {
-                          if (item.onClick) {
-                            close();
+                          if (item.onClick && item.modalContent) {
+                            item.onClick(e);
+                          }
+                          if (item.onClick && !item.modalContent) {
                             item.onClick(e);
+                            close();
```

Comment:
Blir dette litt kortere?
```suggestion
                          if (item.onClick) {
                            item.onClick(e);
                            if (!item.modalContent) {
                              close();
                            }
```